src/app/components/Loop/index.tsx

```diff
@@ -0,0 +1,425 @@
+          <div className="Loop-actions">
+            {/* Don't allow for quote until amount greater than min swap amount is entered and less than max swap amount*/}
+            {this.state.amount !== null &&
+              this.state.amount !== undefined &&
+              parseInt(this.state.amount, 10) > parseInt(loopTerms.min_swap_amount, 10) &&
+              parseInt(this.state.amount, 10) < parseInt(loopTerms.max_swap_amount, 10) &&
+              actions.map((props, idx) => (
+                <Button
+                  key={idx}
+                  {...props}
+                  onMouseOver={
+                    this.state.advanced === true
+                      ? this.props.getLoopOutTerms
+                      : this.props.getLoopInTerms
+                  }
+                  onClick={this.openQuoteModal}
+                />
+              ))}
+          </div>
+          <QuoteModal
+            amt={amount}
+            isOpen={quoteModalIsOpen}
+            onClose={this.openQuoteModal}
+            type={loopType}
+            dest={destination}
+            srf={swapRoutingFee}
+            sf={swapFee}
+            mf={minerFee}
+            pre={prepayAmt}
+            chan={channel}
+            adv={advanced}
+            htlc={htlc}
+            /**
+             * TODO update as needed for future iterations
+             * of loop
+             */
+            sct={this.state.loopType === 'Loop Out' ? conf : ''}
+          />
+        </div>
+      </>
+    );
+  }
+
+  private handleSetChannelId(point: string, openChannels: ChannelWithNode[]) {
+    // Work-around to grab channel id and pass to Loop API
+    const match = openChannels.filter(id => id.channel_point === point);
+    const keys = Object.keys(match);
+    const jsonClone = JSON.parse(JSON.stringify(match));
+    const key = keys[0];
+    const channelId = jsonClone[key].chan_id;
+    message.success('Channel set successfully!');
+    this.setState({ channel: channelId });
+  }
+
+  private handleChangeAmount = (amount: string) => {
+    this.setState({ amount });
+  };
+
+  private handleChangeDestination = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ destination: ev.currentTarget.value });
+  };
+
+  private handleChangeSwapRoutingFee = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ swapRoutingFee: ev.currentTarget.value });
+  };
+
+  private handleChangeSwapFee = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ swapFee: ev.currentTarget.value });
+  };
+
+  private handleChangeMinerFee = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ minerFee: ev.currentTarget.value });
+  };
+
+  private handleChangePrepayAmt = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ prepayAmt: ev.currentTarget.value });
+  };
+
+  private handleChangeConfTarget = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ conf: ev.currentTarget.value });
+  };
```

Comment:
All of these handles could be a single `handleChangeField` method that sets it based on the input's `name` property.

Reply:
Wow, did not know this was possible. Very cool.